cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -170,7 +199,10 @@ func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *core
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "Unable to intepret page token %q: %v", request.GetPaginationOptions().GetPageToken(), err)
 	}
-	charts, _, err := s.manager.GetPaginatedChartListWithFilters(cq, pageOffset, int(pageSize))
+	// The current assetsvc manager works on a page number (ie. 1 for the first page),
+	// rather than an offset.
+	pageNumber := pageOffset + 1
+	charts, _, err := s.manager.GetPaginatedChartListWithFilters(cq, pageNumber, int(pageSize))
```

Comment:
This wasn't so much caching a bug, as refactoring to be clear about the current use of pageNumber (so I could update `pageOffsetFromPageToken` below to return 0 by default, as it should, rather than 1.